internal/hcp/registry/metadata/os_test.go

```diff
@@ -0,0 +1,64 @@
+package metadata
+
+import (
+	"fmt"
+	"runtime"
+	"testing"
+)
+
+// MockExecutor is a mock implementation of CommandExecutor.
+type MockExecutor struct {
```

Comment:
I see now why you created the interface. Makes sense if you want to mock to validate the logic!